example/file_inference/src/audio_reading.rs

```diff
@@ -0,0 +1,192 @@
+        let packet = match format.next_packet() {
+            Ok(packet) => packet,
+            Err(_) => {
+                break;
+            }
+        };
+
+        // If the packet does not belong to the selected track, skip it.
+        if packet.track_id() != track_id {
+            continue;
+        }
+
+        // Decode the packet into audio samples, ignoring any decode errors.
+        match decoder.decode(&packet) {
+            Ok(audio_buf) => {
+                // The decoded audio samples may now be accessed via the audio buffer if per-channel
+                // slices of samples in their native decoded format is desired. Use-cases where
+                // the samples need to be accessed in an interleaved order or converted into
+                // another sample format, or a byte buffer is required, are covered by copying the
+                // audio buffer into a sample buffer or raw sample buffer, respectively. In the
+                // example below, we will copy the audio buffer into a sample buffer in an
+                // interleaved order while also converting to a f32 sample format.
+
+                // If this is the *first* decoded packet, create a sample buffer matching the
+                // decoded audio buffer format.
+                if sample_buf.is_none() {
+                    // Get the audio buffer specification.
+                    let spec = *audio_buf.spec();
+                    original_sample_rate = spec.rate;
+                    num_of_channels = spec.channels.count();
+                    debug!(
+                        "Original file has {} channel(s) with sample rate {}",
+                        num_of_channels, original_sample_rate
+                    );
+
+                    // Get the capacity of the decoded buffer. Note: This is capacity, not length!
+                    let duration = audio_buf.capacity() as u64;
+
+                    // Create the f32 sample buffer.
+                    sample_buf = Some(SampleBuffer::<f32>::new(duration, spec));
+                }
+
+                // Copy the decoded audio buffer into the sample buffer in an interleaved format.
+                if let Some(buf) = &mut sample_buf {
+                    buf.copy_interleaved_ref(audio_buf);
+                    interleave_samples.extend_from_slice(buf.samples());
+                }
+            }
+            Err(Error::DecodeError(_)) => (),
+            Err(Error::IoError(e)) => {
+                dbg!(e);
+                break;
+            }
+            Err(_) => break,
+        }
+    }
+
+    // Now we have interleaved audio, need to convert it to channel based.
+    let stereo_audio = interleaved_to_channel(interleave_samples, num_of_channels);
+
+    // Silero VAD only work with mono audio.
+    let mono_audio = stereo_to_mono(stereo_audio, multi_channel_strategy);
+
+    // Silero VAD only work with 8000 or 16000 Hz audio.
+    resample_pcm(
```

Comment:
Wouldn't it be nicer to just let the resampling happen inside the `VadSession` like the code allows? This example might make people think they have to resample and pass in resampled audio.

Reply:
done